fe/fe-core/src/main/java/com/starrocks/planner/NestLoopJoinNode.java

```diff
@@ -2,32 +2,73 @@
 
 package com.starrocks.planner;
 
-import com.google.common.base.Joiner;
 import com.google.common.base.Preconditions;
 import com.starrocks.analysis.Expr;
 import com.starrocks.analysis.JoinOperator;
+import com.starrocks.analysis.SlotRef;
 import com.starrocks.analysis.TableRef;
+import com.starrocks.common.IdGenerator;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.qe.SessionVariable;
 import com.starrocks.thrift.TNestLoopJoinNode;
 import com.starrocks.thrift.TPlanNode;
 import com.starrocks.thrift.TPlanNodeType;
 import org.apache.commons.collections.CollectionUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.util.ArrayList;
 import java.util.List;
 import java.util.stream.Collectors;
 
 /**
  * NESTLOOP JOIN
  *  TODO: support all kinds of join type
  */
-public class NestLoopJoinNode extends JoinNode {
+public class NestLoopJoinNode extends JoinNode implements RuntimeFilterBuildNode {
+
     private static final Logger LOG = LogManager.getLogger(NestLoopJoinNode.class);
 
     public NestLoopJoinNode(PlanNodeId id, PlanNode outer, PlanNode inner, TableRef innerRef,
                             JoinOperator joinOp, List<Expr> eqJoinConjuncts, List<Expr> joinConjuncts) {
         super("NESTLOOP JOIN", id, outer, inner, joinOp, eqJoinConjuncts, joinConjuncts);
+    }
+
+    @Override
+    public void buildRuntimeFilters(IdGenerator<RuntimeFilterId> generator) {
+        SessionVariable sessionVariable = ConnectContext.get().getSessionVariable();
+        JoinNode.DistributionMode distributionMode = JoinNode.DistributionMode.BROADCAST;
+        PlanNode buildStageNode = this.getChild(1);
+        List<Expr> conjuncts = new ArrayList<>(otherJoinConjuncts);
+        conjuncts.addAll(getConjuncts());
+        for (int i = 0; i < conjuncts.size(); i++) {
+            Expr expr = conjuncts.get(i);
+            // we only support BinaryPredicate to build RuntimeFilter
```

Comment:
1. Checked
2. The constant expression would be pushdown, so it would not be taken into consider
3. pushdown would check if left expression could be bound by left child

Reply:
you should elaborate some cases to see if performance regression happens for case 2, it is a critical issue. t < max_value and t >min_value is always true, selectivity = 0, it should be used.